[basic-sort] sorting added; package updates, cleanup

jtrein opened the pull request:
Recent updates to Blueprint Table and Core have brought improved table perf. I tested their old version with slightly more rows and columns and it was 🐢..

* Updates sort icon (no extraneous styles needed from Material UI) to be simple SVG
* Adds basic sorting: ASC, DESC
* Package cleanup from template (fetch, async, regenerator, etc.)
* Updates blueprint core and table, react bump

I briefly tested on local Universal Codes and it's 👌🏽 .

jtrein commented:
FYI, want to use the updated table for mSupply Customer module, so getting this in now.

jtrein commented:
Chris, if you have time this week, could you review this one? We need it for the msupply-customer module.

Chris-Petty commented:
Whoa sorry. I was in TL when you made this, just back in when you last commented. Am back from surgery now.

Chris-Petty commented on src/SussolReactTable.jsx:
```diff
@@ -2,8 +2,25 @@ import React from 'react';
+*/
+const compare = (a, b, isAscending) => (
+  isAscending
+  ? a.toString().localeCompare(b, undefined, { numeric: true, sensitivity: 'base' })
```
localeCompare built into Node? I gotta take note of this one

jtrein replied:
Those mental JS spec people!

Chris-Petty commented on src/SussolReactTable.jsx:
```diff
@@ -42,24 +81,27 @@ export class SussolReactTable extends React.Component {
+          onClick={
+            column.sortable
+            ? () => this.toggleSortOrder(column.key)
+            : noop
```
null?

jtrein replied:
Probs wrote this too fast. `null` should work, I believe.

jtrein replied:
Just checked. It's b/c Material UI `EnhancedButton` wants a func prop sent down, or it throws.

Chris-Petty commented:
yarn lock conflicts RIP

jtrein commented:
Should be all set to go!

jtrein commented:
* Moved deps on `material-ui` to `peerDependencies`. Otherwise, the bundle becomes bloated in projects where there is already `material-ui` present. The table assumes the dependent is already using a Material-like theme by including it as a design decision.
* Cleaned up some devDeps that are from the template we're not using
* Added a basic test, or two
